proxy/src/dns.rs

```diff
@@ -107,6 +143,14 @@ impl Future for IpAddrFuture {
     }
 }
 
+impl Future for IpAddrListFuture {
```

Comment:
Is there a reason `IpAddrListFuture` is a newtype rather than a type alias? If it were a type alias, the impl of `Future` for it that just calls the wrapped future wouldn't be necessary?

Reply:
> Is there a reason IpAddrListFuture is a newtype rather than a type alias? If it were a type alias, the impl of Future for it that just calls the wrapped future wouldn't be necessary?

I switched it to be the way you recommend. Thanks.